spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -1253,9 +1254,8 @@ object CometSparkSessionExtensions extends Logging {
     org.apache.spark.SPARK_VERSION >= "4.0"
   }
 
-  def usingDataFusionParquetExec(conf: SQLConf): Boolean =
-    Seq(CometConf.SCAN_NATIVE_ICEBERG_COMPAT, CometConf.SCAN_NATIVE_DATAFUSION).contains(
-      CometConf.COMET_NATIVE_SCAN_IMPL.get(conf))
+  def usingDataFusionParquetExec(scanImpl: String): Boolean =
```

Comment:
This refactor has made the call sites more complex/verbose. Before, we had:

```scala
usingDataFusionParquetExec(conf)
```

and now we have

```scala
usingDataFusionParquetExec(CometConf.COMET_NATIVE_SCAN_IMPL.get(conf))
```

I don't see any calls where we pass in a string that didn't come from the conf. @hsiang-c is this refactor really needed?